src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -120,7 +126,16 @@
             throw new IOException("Unsupported credential type: " + credentials.getClass().getName());
         }
 
-        return gb.build();
+        GitHub github = gb.build();
+        try {
+            github.checkApiUrlValidity();
```

Comment:
In github-api client library `FileNotFoundException` can have multiple meanings. This is was only way that I know to verify if github.com is reacheable.

At any rate, let me give it a thought.